lib/Mojo/Base.pm

```diff
@@ -101,6 +105,16 @@ sub tap {
   return $self;
 }
 
+sub with_roles {
+  if ($_[0]->can_roles()) {
+    Carp::croak "class and roles must be package names" if !!grep { ref $_ } @_;
+    return Role::Tiny->create_class_with_roles(@_);
+  }
+  else {
+    Carp::croak "Role::Tiny 2.000005 is required for with_roles method";
```

Comment:
I would recommend at least 2.000001 for an important fix to created class names. https://rt.cpan.org/Public/Bug/Display.html?id=103310